watson/utils.py

```diff
@@ -32,3 +34,17 @@ def sorted_groupby(iterator, key, reverse=False):
     key first.
     """
     return itertools.groupby(sorted(iterator, key=key, reverse=reverse), key)
+
+
+def options(opt_list):
+    """
+    Wrapper for the `value_proc` field in `click.prompt`, which validates
+    that the user response is part of the list of accepted responses.
+    """
+    def value_proc(user_input):
+        if user_input in opt_list:
+            return user_input
+        else:
+            raise UsageError("Response should be one of [{}]".format(
+                ','.join([str(x) for x in opt_list])))
```

Comment:
Thanks! I remember spotting this earlier, but I thought I had changed `join()` to use the generator expression form, but evidently I forgot. It wouldn't hurt to make this change, and it'll make this line look a little more readable.